lib/GuestManager.php

```diff
@@ -69,6 +76,125 @@ public function updateName(Room $room, Participant $participant, string $display
+	 * @throws GuestImportException
+	 */
+	public function importEmails(Room $room, $file, bool $testRun): array {
+		if ($room->getType() === Room::TYPE_ONE_TO_ONE
+			|| $room->getType() === Room::TYPE_ONE_TO_ONE_FORMER
+			|| $room->getType() === Room::TYPE_NOTE_TO_SELF
+			|| $room->getObjectType() === BreakoutRoom::PARENT_OBJECT_TYPE
+			|| $room->getObjectType() === Room::OBJECT_TYPE_VIDEO_VERIFICATION) {
+			throw new GuestImportException(GuestImportException::REASON_ROOM);
+		}
+
+		$content = fopen($file['tmp_name'], 'rb');
+		$details = fgetcsv($content, escape: '');
+		if (!isset($details[0]) || strtolower($details[0]) !== 'email') {
+			throw new GuestImportException(
+				GuestImportException::REASON_HEADER_EMAIL,
+				$this->l->t('Missing email field in header line'),
+			);
+		}
+		if (isset($details[1]) && strtolower($details[1]) !== 'name') {
+			throw new GuestImportException(
+				GuestImportException::REASON_HEADER_NAME,
+				$this->l->t('Missing name field in header line'),
+			);
+		}
+
+		$participants = $this->participantService->getParticipantsByActorType($room, Attendee::ACTOR_EMAILS);
+		$alreadyInvitedEmails = array_flip(array_map(static fn (Participant $participant): string => $participant->getAttendee()->getInvitedCloudId(), $participants));
+
+		$line = $duplicates = 0;
+		$emailsToAdd = $invalidLines = [];
+		while ($details = fgetcsv($content, escape: '')) {
+			$line++;
+			if (isset($alreadyInvitedEmails[$details[0]])) {
+				$this->logger->debug('Skipping import of ' . $details[0] . ' (line: ' . $line . ') as they are already invited');
+				$duplicates++;
+				continue;
+			}
+
+			if (count($details) > 2) {
+				$this->logger->debug('Invalid entry with too many fields on line: ' . $line);
+				$invalidLines[] = $line;
+				continue;
+			}
+
+			if (count($details) === 2) {
+				[$email, $name] = $details;
+			} else {
+				$email = $details[0];
+				$name = null;
+			}
+
+			$email = strtolower($email);
+			if (!$this->validateMailAddress($email)) {
+				$this->logger->debug('Invalid email "' . $email . '" on line: ' . $line);
+				$invalidLines[] = $line;
+				continue;
+			}
+
+			if ($name !== null && strcasecmp($name, $email) === 0) {
+				$name = null;
+			}
+
+			$actorId = hash('sha256', $email);
+			$alreadyInvitedEmails[$email] = $actorId;
+			$emailsToAdd[] = [
+				'email' => $email,
+				'actorId' => $actorId,
+				'name' => $name,
+			];
+		}
+
+		if ($testRun) {
+			if (empty($invalidLines)) {
+				return [
+					'invites' => count($emailsToAdd),
+					'duplicates' => $duplicates,
+				];
+			}
+
+			throw new GuestImportException(
+				GuestImportException::REASON_ROWS,
+				$this->l->t('Following lines are invalid: %s', implode(', ', $invalidLines)),
+				$invalidLines,
+				count($emailsToAdd),
+				$duplicates,
+			);
+		}
+
+		foreach ($emailsToAdd as $add) {
+			$participant = $this->participantService->inviteEmailAddress($room, $add['actorId'], $add['email'], $add['name']);
```

Comment:
I'm not receiving either e-mail invites in mailhog / new entries in DB / participants in the list, but should happened here?

Reply:
Default of the testRun parameter was true, but you don't send the parameter when it's false it seems. I can change the default.

Reply:
- It's actually a bug in server because you DO send the value as false: https://github.com/nextcloud/server/pull/49515
We can work around it by changing the default to false and you not sending the value in that case.

Reply:
Changed here in the API PR and in the frontend PR and it's working now.